### ISSUE TITLE
Should empty attribute names be prevented?

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#2415]
At the moment it is possible to create attributes with empty names, i.e. `""` is not rejected as an attribute name. This is probably not a good idea. Do we want to allow it?

Arguments against:

 - May be difficult to access in some interfaces (does work in R though).
 - Not possible to store in some file format (GML writer does modify it by adding "igraph" to it due to the default prefixing of any attribute name not starting with a letter).
 - Possible to store in file formats, but the resulting file is technically invalid (GraphML).
 - It is an edge case that _might_ need special handling in some cases. We are likely to forget about it.
 - There is just not much use for allowing it (?)

The new Pajek reader now preserves custom parameter names, and accepts the empty string, `""`, as a valid name. I am going to prevent this.

**Comment by szhorvat** [comment]
@ntamas Do you agree, and if yes, can you fix this, either on `master` or `develop` according to your preference?

**Comment by ntamas** [comment]
I'm not a big fan of restricting a data structure on the C side just because certain file formats are not able to cope with certain special cases. I don't even think that there is anything in the GraphML specs that would prevent anyone from creating an attribute with an empty name, is there? Or even if there was -- people could have created graphs with an empty attribute name fairly easily in Python, then saved them to Python's pickle format, and this change would suddenly prevent them from loading these graphs back for no reason.

So, I'd rather introduce restrictions on each file format exporter instead of preventing empty attribute names straight away.

**Comment by szhorvat** [comment]
Alright, then we come back to this if things blow up ...

**Comment by szhorvat** [comment]
> I don't even think that there is anything in the GraphML specs that would prevent anyone from creating an attribute with an empty name, is there?

I think there is. 

http://graphml.graphdrawing.org/specification/xsd.html#element-key

`id` should be an `xs:NMTOKEN`, whatever that is.

https://www.oreilly.com/library/view/xml-schema/0596002521/re84.html

That's defined as `\c+`, which I assume means "one or more characters", not zero or more.

That said, igraph does read such GraphML files, see #2505.

I tried it with yEd, and it crashed. 

I also tried it with Gephi, and it gave an error: "Attribute parse error, id is missing. SEVERE"
